src/applications/vaos/appointment-list/components/AppointmentsPage/ScheduleNewAppointmentRadioButtons.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import RadioButtons from '@department-of-veterans-affairs/component-library/RadioButtons';
+import { Link } from 'react-router-dom';
+import recordEvent from 'platform/monitoring/record-event';
+import { GA_PREFIX } from 'applications/vaos/utils/constants';
+
+export default function ScheduleNewAppointmentRadioButtons({
+  showCheetahScheduleButton,
+  startNewAppointmentFlow,
+}) {
+  const [radioSelection, setRadioSelection] = useState();
+  const [radioSelectionEvent, setRadioSelectionEvent] = useState();
+
+  function radioOptions() {
+    const optionsArray = [
+      {
+        value: 'new-appointment',
+        label: 'Primary or specialty care',
+      },
+    ];
+
+    if (showCheetahScheduleButton) {
+      optionsArray.push({
+        value: 'new-covid-19-vaccine-booking',
+        label: 'COVID-19 vaccine',
+      });
+    }
+    return optionsArray;
+  }
+
+  return (
+    <>
+      <h2 className="vads-u-padding-bottom--0">Schedule a new appointment</h2>
+      <RadioButtons
+        name={'schedule-new-appointment'}
+        id={'schedule-new-appointment'}
+        options={radioOptions()}
+        additionalFieldsetClass="vads-u-margin-top--0"
+        onValueChange={({ value }) => {
+          setRadioSelection(value);
+          setRadioSelectionEvent(
```

Comment:
It looks like you could determine the event name based on the selected value, so you don't necessarily need a second state value here.

Generally, React encourages you to set the minimum amount of state that you need and then derive any additional data based on that when you need it.

Reply:
Removed, good call. I was thinking better to keep it up in a state variable rather than have convoluted ternary operations. Thanks for the feedback 😄